ConcurrentPercolator/ConcurrentPercolator.tla

```diff
@@ -102,8 +115,51 @@ checkSnapshotIsolation(k, commit_ts) ==
   THEN
     key_si' = [key_si EXCEPT ![k] = FALSE]
   ELSE
-    UNCHANGED <<key_si>>
-
+    UNCHANGED <<key_si>> 
+
+\* Returns index of $e$ in $seq$ 
+getIndex(seq, e) == CHOOSE n \in DOMAIN seq : seq[n] = e  
+
+\* Returns the write whose $ts$ is largest
+getWriteWithMaxTS(w) == CHOOSE x \in w : \A y \in w : x.ts >= y.ts
+
+\* Returns the writes with ts less or equal than $ts$ 
+findWriteLessEqualTS(kw, ts) ==
+  {w \in Range(kw) : (w.ts <= ts /\ IF (w.type = "rollback" /\ w.is_deleted = TRUE) THEN FALSE ELSE TRUE)}
```

Comment:
is `{w \in Range(kw) : (w.ts <= ts /\ w.is_deleted = FALSE)}` enough?

Reply:
anyway why use `if then else`, just use  `(w.type = "write" \/ w.is_deleted = FALSE)`